### ISSUE TITLE
BWA: stat: invalid option -- 's'

### ISSUE THREAD

**Issue by scholtalbers** [galaxyproject/tools-devteam#81]
On bwa and bwa-mem the error at the end of this ticket appears in the stderr. Process seems to finish correctly.
```
Galaxy Tool ID:	toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa/0.1
Revision: c71dd035971e
```

Job Command-Line:
```bash
  ln -s "/g/galaxy/galaxy-dev_data/files/009/dataset_9059.dat" "localref.fa" && ( size=`stat -c %s "localref.fa" 2>/dev/null`; if [ $? -eq 0 ]; then if [ $size -lt 2000000000 ]; then bwa index -a is "localref.fa"; else bwa index -a bwtsw "localref.fa"; fi; fi; eval $(stat -s "localref.fa"); if [ $? -eq 0 ]; then if [ $st_size -lt 2000000000 ]; then bwa index -a is "localref.fa"; echo "Generating BWA index with is algorithm"; else bwa index -a bwtsw "localref.fa"; echo "Generating BWA index with bwtsw algorithm"; fi; fi; ) && bwa aln -t "${GALAXY_SLOTS:-1}" -b -0 "localref.fa" "/g/galaxy/galaxy-dev_data/files/008/dataset_8792.dat" > first.sai && bwa samse "localref.fa" first.sai "/g/galaxy/galaxy-dev_data/files/008/dataset_8792.dat" | samtools view -Sb - > temporary_bam_file.bam && samtools sort -f temporary_bam_file.bam /g/galaxy/galaxy-dev_data/files/009/dataset_9061.dat
```
```
Input Parameter: Value Note for rerun  
Load reference genome from 	history 	
Use the following dataset as the reference sequence 	1: UCSC Main on Mouse: refGene (chr12:1-121257530) 	
Select input type: single_bam 	
Select BAM dataset: 2: Downsample_KO 	
Set advanced single end options?: do_not_set 	
Set readgroups information?: do_not_set 	
Select analysis mode: illumina 
```

***Error***:
```
[bwa_index] Pack FASTA... 0.58 sec
[bwa_index] Construct BWT for the packed sequence...
[bwa_index] 23.55 seconds elapse.
[bwa_index] Update BWT... 0.40 sec
[bwa_index] Pack forward-only FASTA... 0.34 sec
[bwa_index] Construct SA from BWT and Occ... 11.21 sec
[main] Version: 0.7.10-r876-dirty
[main] CMD: bwa index -a is localref.fa
[main] Real time: 38.455 sec; CPU: 36.091 sec
stat: invalid option -- 's'
Try `stat --help' for more information.
/g/galaxy/galaxy-dev_data/job_working_directory/005/5943/galaxy_5943.sh: line 35: [: -lt: unary operator expected
[bwa_index] Pack FASTA... 0.54 sec
[bwa_index] Construct BWT for the packed sequence...
[BWTIncCreate] textLength=127596972, availableWord=20977816
[BWTIncConstructFromPacked] 10 iterations done. 34603372 characters processed.
[BWTIncConstructFromPacked] 20 iterations done. 63925708 characters processed.
[BWTIncConstructFromPacked] 30 iterations done. 89983420 characters processed.
[BWTIncConstructFromPacked] 40 iterations done. 113139548 characters processed.
[bwa_index] 56.18 seconds elapse.
[bwa_index] Update BWT... 0.38 sec
[bwa_index] Pack forward-only FASTA... 0.34 sec
[bwa_index] Construct SA from BWT and Occ... 11.34 sec
[main] Version: 0.7.10-r876-dirty
[main] CMD: bwa index -a bwtsw localref.fa
[main] Real time: 70.975 sec; CPU: 68.787 sec
[W::sam_hdr_parse] duplicated sequence 'mm9_refGene_NM_011899'
[W::sam_hdr_parse] duplicated sequence 'mm9_refGene_NM_001100116'
[W::sam_hdr_parse] duplicated sequence 'mm9_refGene_NM_028527'
[W::sam_hdr_parse] duplicated sequence 'mm9_refGene_NM_001177573'
[W::sam_hdr_parse] duplicated sequence 'mm9_refGene_NM_001166062'
[W::sam_hdr_parse] duplicated sequence 'mm9_refGene_NM_178657'
[main] Version: 0.7.10-r876-dirty
[main] CMD: bwa mem -t 1 -v 1 localref.fa /g/galaxy/galaxy-dev_data/files/009/dataset_9065.dat
[main] Real time: 47.632 sec; CPU: 39.081 sec
```

**Comment by nsoranzo** [comment]
If you're running under Linux and the output datasets are green, you don't need to worry, this is exactly how it is supposed to behave. In fact, the cited `<command>` contains shell code needed to support the different syntax of the ```stat``` command under Linux and OS X.

**Comment by scholtalbers** [comment]
hmm, I think there still might be an issue here as st_size never seems to be initialized?([: -lt: unary operator expected)
Thus maybe it could read something like this, to remove both error messages?
```
st_stat=$(stat -s "localref.fa" 2> /dev/null)
```

**Comment by nsoranzo** [comment]
No, ```eval $(stat -s "localref.fa")``` under OS X creates some environment variables and ```st_stat``` is one of them. The problem is that ```$?``` of this command should be 1 under Linux, but it seems it is 0 instead.

Just to confirm, is your Galaxy running under Linux and are the output datasets green?

**Comment by scholtalbers** [comment]
Yes, running Linux and get green outputs.

**Comment by nsoranzo** [comment]
@scholtalbers If you have time, you may test the above-mentioned commit and see if it gets better.

**Comment by scholtalbers** [comment]
Almost, runs fine on Linux if the following is applied (or the "if illumina" completely removed): https://gist.github.com/scholtalbers/f2c653d08e85e0a00b2d

**Comment by nsoranzo** [comment]
Thanks, this is fixed in 800aa3b2fd0b3cbd46fe62ecebbc4527399f8ec7 . I'm going to merge the other commit and close this issue if it's fine for you.

**Comment by scholtalbers** [comment]
yes and thanks!
